pkg/collector/corechecks/cluster/orchestrator/processors/processor.go

```diff
@@ -165,16 +165,19 @@ func (p *Processor) Process(ctx *ProcessorContext, list interface{}) (processRes
 		})
 	}
 
-	// Split messages in chunks for orchestrator metadata and manifest data.
-	chunkCount := orchestrator.GroupSize(len(resourceMetadataModels), ctx.Cfg.MaxPerMessage)
-	metadataChunks := chunkResources(resourceMetadataModels, chunkCount, ctx.Cfg.MaxPerMessage)
-	manifestChunks := chunkResources(resourceManifestModels, chunkCount, ctx.Cfg.MaxPerMessage)
-	metadataMessages := make([]model.MessageBody, 0, chunkCount)
-	manifestMessages := make([]model.MessageBody, 0, chunkCount)
+	// Split messages in chunks
+	metadataChunker := &collectorOrchestratorChunker{}
+	chunkOrchestratorPayloadsBySizeAndWeight(resourceMetadataModels, resourceManifestModels, ctx.Cfg.MaxPerMessage, ctx.Cfg.MaxWeightPerMessageBytes, metadataChunker)
+	manifestChunker := &collectorOrchestratorChunker{}
+	chunkOrchestratorPayloadsBySizeAndWeight(resourceManifestModels, resourceManifestModels, ctx.Cfg.MaxPerMessage, ctx.Cfg.MaxWeightPerMessageBytes, manifestChunker)
```

Comment:
I'm pretty confused by the implementation. If I understand properly the idea is to chunk resources based on the serialized size of their manifest. If that is correct I would advise to make it clear in the code with extra comments and in the PR description.

Why do we call that function with the same arguments twice?

As I see it, `orchestratorList` contains two different slices, one for metadata payloads and one for manifest payloads but its `ToChunk()` method only takes action on the metadata one.

Where are manifest payloads chunked now?

Reply:
Thanks for the feedback, your understand is correct.

The function is called twice but with different first argument value (`resourceMetadataModels` and  `resourceManifestModels`), the first call is to chunk metadata, the second call is to chunk manifest.

The second argument of `chunkOrchestratorPayloadsBySizeAndWeight` is used to know the serialized size of manifest. When chunking manifest itself, the first argument and the second one should be the same.

I can add more comments on it

Reply:
oh right misread the first parameter 🤦
that makes sense now

I think adding extra comments on the fact `orchestratorPayloads` is a generic argument for metadata or manifest payloads would be nice to the reader